### PR TITLE
Revert to automatic handling of signing

### PR DESCRIPTION
**Pull request by Shelnutt2**
The manual signing process wasn't triggering properly, as a result the maven central uploads were being rejected from being published. They were accepted for upload but the releases always were erroring out on closing.

This PR reverts to the previous method of automatically handling signing and include the simple fix that was needed to allow publishing to local maven repo without signing.

As a test, builds from this branch were temporarily allowed to upload snapshots and you can see here that the signing was successful and upload worked:
https://travis-ci.org/TileDB-Inc/TileDB-Java/jobs/574431737

### CODE REVIEW

**Comment by Shelnutt2** [comment]
@jakebolewski can you test this branch to make sure you are able to run `publishToMavenLocal` locally to publish to local repository without signing? I've test this on my machine and an ec2 instance and it looks like its working fine.

File: build.gradle [hunk 1]
```diff
@@ -139,6 +139,20 @@ artifacts {
...
 
+import org.gradle.plugins.signing.Sign
+gradle.taskGraph.whenReady { taskGraph ->
+    if (taskGraph.allTasks.any { it instanceof Sign }) {
+        allprojects { ext."signing.keyId" = System.getenv('GPG_KEY_ID') }
+        allprojects { ext."signing.secretKeyRingFile" = System.getenv('GPG_KEY_LOCATION') }
+        allprojects { ext."signing.password" = System.getenv('GPG_PASSPHRASE') }
+    }
+    // Do not sign archives by default (a local build without gpg keyring should succeed)
+    if (taskGraph.allTasks.any { it.name == 'build' || it.name == 'assemble'  || it.name == 'publishToMavenLocal'}) {
```
Comment (jakebolewski) [inline build.gradle]: could you remove this and add:

```
ext.isReleaseVersion = getProjectProperty('TILEDB_JAVA_RELEASE').toBoolean()
tasks.withType(Sign) {
    onlyIf { isReleaseVersion }
}
```

I think it's effectively doing the same thing, add allprojects to the closure might not be correct.

**Comment by jakebolewski** [comment]
lgtm, although if you call a task that is not manually filtered will it enable signing?

File: build.gradle (hunk 1 above)
Comment (Shelnutt2) [reply build.gradle]: The release properties was what was not trigger the signing before, and I tested some changes to with no success. You can see for snapshots and for releases it's skipping the signing:
`Task :signMavenJavaPublication SKIPPED`

https://travis-ci.org/TileDB-Inc/TileDB-Java/jobs/574055440
https://travis-ci.org/TileDB-Inc/TileDB-Java/jobs/574370424

**Comment by Shelnutt2** [comment]
> lgtm, although if you call a task that is not manually filtered will it enable signing?

Yes, is there a task you use that is not on the list? This covers everything used by ci and what is in the readme.

**Pull request merged**